### PR TITLE
Bluetooth: shell: Fix PER SYNC conditional compilation 

### PR DESCRIPTION
**Pull request by cvinayak**
Fix conditional compilation to allow building test shell
with Periodic Advertising Sync alone.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/shell/bt.c [hunk 1]
```diff
@@ -1377,7 +1377,11 @@ static int cmd_per_adv_data(const struct shell *shell, size_t argc,
 	return 0;
 }
 #endif /* CONFIG_BT_PER_ADV */
+#endif /* CONFIG_BT_EXT_ADV */
+#endif /* CONFIG_BT_BROADCASTER */
 
+#if defined(CONFIG_BT_OBSERVER)
+#if defined(CONFIG_BT_EXT_ADV)
 #if defined(CONFIG_BT_PER_ADV_SYNC)
```
Comment (Thalley) [inline subsys/bluetooth/shell/bt.c:1385]: `BT_PER_ADV_SYNC` depends on `BT_OBSERVER` and cannot be set if `BT_EXT_ADV` isn't set. In that sense, I suppose we could reduce this to simply be
```
#if defined(CONFIG_BT_PER_ADV_SYNC)
```

File: subsys/bluetooth/shell/bt.c [hunk 2]
```diff
@@ -2781,7 +2785,11 @@ SHELL_STATIC_SUBCMD_SET_CREATE(bt_cmds,
 		      "[<interval-min> [<interval-max> [tx_power]]]",
 		      cmd_per_adv_param, 1, 3),
 	SHELL_CMD_ARG(per-adv-data, NULL, "<data>", cmd_per_adv_data, 2, 0),
-#endif
+#endif /* CONFIG_BT_PER_ADV */
+#endif /* CONFIG_BT_EXT_ADV */
+#endif /* CONFIG_BT_BROADCASTER */
+#if defined(CONFIG_BT_OBSERVER)
+#if defined(CONFIG_BT_EXT_ADV)
 #if defined(CONFIG_BT_PER_ADV_SYNC)
```
Comment (Thalley) [inline subsys/bluetooth/shell/bt.c:2793]: Ditto

**Review by Thalley — CHANGES_REQUESTED** [review]

**Comment by joerchan** [comment]
The commit title shouldn't have "tests:" in it. The change is in the bluetooth shell subystem.

**Review by jhedberg — APPROVED** [review]

**Review by Thalley — APPROVED** [review]
LGTM

**Review by joerchan — APPROVED** [review]

**Pull request merged**